### PR TITLE
Update go.mod and drop patch version in CHANGELOG.md

### PR DESCRIPTION
**Pull request by daxmc99**
Fixes https://github.com/sourcegraph/sourcegraph/issues/18026

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/sourcegraph/sourcegraph
 
-go 1.14
+go 1.15
```
Comment (keegancsmith) [inline go.mod:3]: FYI you don't have to bump this, will just prevent someone from using older versions of go on our codebase (which seems unneccessary?) From `go help go.mod`

> The expected language version, set by the go directive, determines
which language features are available when compiling the module.
Language features available in that version will be available for use.
Language features removed in earlier versions, or added in later versions,
will not be available. Note that the language version does not affect
build tags, which are determined by the Go release being used.

File: CHANGELOG.md [hunk 2]
```diff
@@ -28,10 +28,16 @@ All notable changes to Sourcegraph are documented in this file.
...
-  - RDS Customers please reference [AWS' documentation on updating the SSL/TLS certificate](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL-certificate-rotation.html).
+- Secrets (such as access tokens and passwords) will now appear as REDACTED when editing external service config, and in
+  graphql API responses. [#17261](https://github.com/sourcegraph/sourcegraph/issues/17261)
+- Sourcegraph is now built with Go 1.15
+  - Go `1.15` introduced changes to SSL/TLS connection validation which requires certificates to include a `SAN`. This
+    field was not included in older certificates and clients relied on the `CN` field. You might see an error
+    like `x509: certificate relies on legacy Common Name field`. We recommend that customers using Sourcegraph with an
+    external database and and connecting to it using SSL/TLS check whether the certificate is up to date.
+  - RDS Customers please
+    reference [AWS' documentation on updating the SSL/TLS certificate](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL-certificate-rotation.html)
```
Comment (keegancsmith) [inline CHANGELOG.md:39]: minor: I think our markdown style is to not reflow text (ie add in new lines). I think CI used to enforce this, but not sure if it still does.

**Review by keegancsmith — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (daxmc99) [reply CHANGELOG.md]: This was a bad editor config on my part, prettier does enforce some formatting but it doesn't perform the reflowing

File: go.mod (hunk 1 above)
Comment (daxmc99) [reply go.mod:3]: My thoughts were that it would prompt more devs to use `1.15` which might hit bugs before the customer release on the 20th.

**Pull request merged**